kasa/tapo/tapobulb.py

```diff
@@ -17,6 +19,17 @@ class TapoBulb(TapoDevice, SmartBulb):
     Documentation TBD. See :class:`~kasa.smartbulb.SmartBulb` for now.
     """
 
+    def __init__(
+        self,
+        host: str,
+        *,
+        port: Optional[int] = None,
+        credentials: Optional[Credentials] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host, port=port, credentials=credentials, timeout=timeout)
+        self._device_type = DeviceType.TapoBulb
```

Comment:
Let's refrain from us for now. I think we should sit down and decide upon the common interfaces. IMO, the internal "device_type" should become something that describes the device type (i.e., which interface(s) are accessible) w/o .

Using `tapoplug` and `tapobulb` as shortcuts have been fine, and maybe we should still allow doing that for the cli for the time being, but I'd rather avoid these in the exposed API; the end user should not have to care about the internal impl. details that much, bulbs implement a bulb interface, plugs implemented a plug interface etc. and that's about it.

Reply:
Maybe we should just ignore the `device_type` parametrization as a mishap and make downstreams use the new connection parameters this PR is introducing, or could that cause some issues?

The only reason for the `device_type` is basically to inform about the interface that the device provides to let downstreams figure out how they interact with the device ("can I use bulb commands?", ..). That is the only thing downstreams should be concerned about, everything what happens under the hood behind the device type specific interface shouldn't matter.

Reply:
device_type parameterization now removed